perf/reader.go

```diff
@@ -222,34 +199,23 @@ func NewReaderWithOptions(array *ebpf.Map, perCPUBuffer int, opts ReaderOptions)
 		rings = append(rings, ring)
 		pauseFds = append(pauseFds, ring.fd)
 
-		if err := addToEpoll(epollFd, ring.fd, len(rings)-1); err != nil {
+		if err := poller.Add(ring.fd, i); err != nil {
 			return nil, err
 		}
 	}
 
-	closeFd, err := unix.Eventfd(0, unix.O_CLOEXEC|unix.O_NONBLOCK)
-	if err != nil {
-		return nil, err
-	}
-	fds = append(fds, closeFd)
-
-	if err := addToEpoll(epollFd, closeFd, -1); err != nil {
-		return nil, err
-	}
-
 	array, err = array.Clone()
 	if err != nil {
 		return nil, err
 	}
 
 	pr = &Reader{
-		array:   array,
-		rings:   rings,
-		epollFd: epollFd,
+		array:  array,
+		rings:  rings,
+		poller: poller,
 		// Allocate extra event for closeFd
```

Comment:
This should probably say "extra event for event fd". I'm thinking would it be better to hide this extra "+1" `EpollEvent` from `Poller` users? In the ringbuf's `ePollEvents` the extra "+1" is not added but AFAICS it's currently needed.

Reply:
Yeah, good point. FWIW I think the extra event is not needed:
* We use the event fd to unblock Read(), so that Close can acquire the mutex
* Without the `+1` event we're not guaranteed to receive the close event, but we are guaranteed to be unblocked if any event is pending

Reply:
Agree, makes sense.